rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -670,6 +670,7 @@ private BlockResult executeForMiningAfterRSKIP144(
             loggingExecuteTxAndReceipt(block, i, tx);
 
             i++;
+            txindex++;
```

Comment:
txindex must be incremented also if we discard the tx (right before the `continue` statement at line 628)